src/_gettsim/functions/loader.py

```diff
@@ -165,91 +221,104 @@ def _convert_path_to_module_name(path: Path, package_root: Path) -> str:
     )
 
 
-def _create_policy_function_from_decorated_callable(
-    function: Callable,
-    module_name: str,
-) -> PolicyFunction:
+def _convert_system_path_to_tree_path(
+    system_path: Path, package_root: Path
+) -> tuple[str, ...]:
     """
-    Create a policy function from a callable with a `@policy_info` decorator.
+    Convert a system path to a tree path.
+
+    The system path is the path to the python module on the user's system. The tree path
+    are the branches of the tree.
 
     Parameters
     ----------
-    function:
-        The callable to wrap.
-    module_name:
-        The name of the module in which the callable is defined.
+    system_path:
+        The path to the python module on the user's system.
+    package_root:
+        The root of the package that contains the functions.
 
     Returns
     -------
-    policy_function:
-        The policy function.
-    """
-
-    # Only needed until the directory structure is cleaned up
-    clean_module_name = (
-        module_name.removeprefix("_gettsim.")
-        .removeprefix("taxes.")
-        .removeprefix("transfers.")
-    )
+    tuple[str, ...]:
+        The tree path.
 
-    return PolicyFunction(
-        function=function,
-        module_name=clean_module_name,
+    Examples
+    --------
+    >>> _convert_system_path_to_tree_path(RESOURCE_DIR / "taxes" / "dir"
+        / "functions.py")
+    ("dir", "functions")
+    """
+    # TODO(@MImmesberger): Remove the removeprefix calls once directory structure is
+    #  changed.
+    branches = tuple(
+        system_path.relative_to(package_root.parent)
+        .with_suffix("")
+        .as_posix()
+        .removeprefix("_gettsim/")
+        .removeprefix("taxes/")
+        .removeprefix("transfers/")
+        .split("/")
     )
-
-
-def _is_function_defined_in_module(function: Callable, module: ModuleType) -> bool:
-    """Check if a function is defined in a specific module or only imported."""
-    return function.__module__ == module.__name__
+    return _simplify_tree_path_when_module_name_equals_dir_name(branches)
 
 
 _AggregationVariant: TypeAlias = Literal["aggregate_by_group", "aggregate_by_p_id"]
 
 
-def load_internal_aggregation_dict(variant: _AggregationVariant):
-    """
-    Load a dictionary with all aggregations by group or person that are defined for
-    internal functions.
+def load_internal_aggregation_tree(
```

Comment:
```suggestion
def load_one_aggregation_specs_tree(
```

And use that terminology throughout, as we already do in `PolicyEnvironment`.

Be consistent in the naming there, too: 

`_aggregate_by_group_specs` -> `_by_group_aggregation_specs_tree` etc.

Reply:
There is only one aggregation specs tree, so I wouldn't choose this function name. If you want to reflect the fact that this is not the final (augmented) tree, we could use `load_aggregations_tree_from_internal_modules`. But I think the status quo is fine as well.